### PR TITLE
Move s:viml_builtin_functions to s:VimLParser.builtin_functions

### LINKED ISSUE #118: Move s:viml_builtin_functions to shared file?
Currently `s:viml_builtin_functions` is defined twice in both `py/pycompiler.vim` (https://github.com/vim-jp/vim-vimlparser/blob/edd6a79ebf52930452aeaec13e5a6f9423896379/py/pycompiler.vim#L825) and `js/jscompiler.vim`.

I think this should be shared, probably in `s:VimLParser.builtin_functions`.

It is likely also out of date.  It would be good if this could be used through some Vim function directly.

### PR DESCRIPTION
**Pull request by tyru**
Fix #118

### CODE REVIEW

**Comment by tyru** [comment]
@blueyed thanks for review!
@mattn to make sure, please review this

File: scripts/update_builtin_functions.vim [hunk 1]
```diff
@@ -0,0 +1,72 @@
+" create builtin function table
+
+" TODO more loose pattern?
+function! s:get_parse_lines(lines) abort
+  let from = index(a:lines, '} functions[] =')
+  if from ==# -1
+    return []
```
Comment (mattn) [inline scripts/update_builtin_functions.vim]:
```
throw 'cannot parse functions'
```

Comment (tyru) [reply scripts/update_builtin_functions.vim]: agree. `update_builtin_commands.vim` also has the same code. thanks!

Comment (tyru) [reply scripts/update_builtin_functions.vim]: fixed fc35a19c1eafb15d17e8fafbf2e6dfd7e0cc5e56

**Comment by mattn** [comment]
LGTM～

**Comment by mattn** [comment]
Thanks!

**Pull request merged**